ui-tests/src/main/java/io/jenkins/plugins/coverage/CoveragePublisher/Adapter.java

```diff
@@ -0,0 +1,99 @@
+package io.jenkins.plugins.coverage.CoveragePublisher;
+
+import org.jenkinsci.test.acceptance.po.Control;
+import org.jenkinsci.test.acceptance.po.PageArea;
+import org.jenkinsci.test.acceptance.po.PageAreaImpl;
+
+import io.jenkins.plugins.coverage.CoveragePublisher.Threshold.AdapterThreshold;
+import io.jenkins.plugins.coverage.CoveragePublisher.Threshold.AdapterThreshold.AdapterThresholdTarget;
+
+/**
+ * Adapter which can be added in the configuration of the {@link CoveragePublisher} of a FreeStyle Project.
+ */
+public class Adapter extends PageAreaImpl {
+    private final Control reportFilePath = control("path");
+    private final Control threshold = control("repeatable-add");
+    private final Control mergeToOneReport = control("mergeToOneReport");
+
+    private final Control delete = control("repeatable-delete");
+    private final Control advancedOptions = control("advanced-button");
+
+    /**
+     * Constructor to create {@link Adapter} for {@link CoveragePublisher}.
+     *
+     * @param reportPublisher
+     *         which should be created, f. e. jacoco or cobertura
+     * @param path
+     *         of parent page
+     */
+    public Adapter(PageArea reportPublisher, String path) {
+        super(reportPublisher, path);
+    }
+
+    /**
+     * Setter for path of report file.
+     *
+     * @param reportFilePath
+     *         path to report file.
+     */
+    public void setReportFilePath(String reportFilePath) {
```

Comment:
A good pattern for the `setters` would be to return the page object so that one can write
```
new PageObject().doSomething().setOne().setTwo()...
```